pkg/chunk/chunk_store.go

```diff
@@ -205,15 +205,19 @@ func (c *store) LabelValuesForMetricName(ctx context.Context, userID string, fro
 	}
 
 	var result []string
+	values := map[string]struct{}{}
 	for _, entry := range entries {
 		_, labelValue, _, _, err := parseChunkTimeRangeValue(entry.RangeValue, entry.Value)
 		if err != nil {
 			return nil, err
 		}
-		result = append(result, string(labelValue))
+		v := string(labelValue)
+		if _, ok := values[v]; !ok {
+			values[v] = struct{}{}
+			result = append(result, v)
+		}
+
 	}
-	sort.Strings(result)
```

Comment:
To my understanding, the changes in this function are to remove the sorting, because - according to the rest of the changes - values will be sorted later on in the composite store. Am I correct?

If correct, it looks weird to me that `LabelValuesForMetricName()` in the chunk store and composite store have a different behaviour. The first one returns non-sorted values, while the second one does. However both implements the same `Store` interface and - by contract - should behave the same. Thoughts?

Reply:
yes you're correct that was my idea. If we're not worried about performance, I'll add back sorting and uniquess on each storage.

Reply:
We are worried about performance, but worried more about correctness.
It is necessary to highlight inconsistencies and clearly explain in comments why it's ok to break a contract or an invariant.